src/libpmemfile-posix/inode.h

```diff
@@ -48,6 +48,12 @@ struct pmemfile_vinode {
 	/* read-write lock, also protects inode read/writes */
 	os_rwlock_t rwlock;
 
+	/*
+	 * Counter to keep track of modifications that potentially
+	 * invalidate a block_pointer_cache field in pmemfile_file struct.
+	 */
+	uint64_t block_pointer_invalidation_counter;
```

Comment:
It seems you are modifying this only in fallocate and truncate.

Why not do it in vinode_allocate_interval / vinode_remove_interval and only when block_array layout changes (any *existing* block_desc changes position)?